zebrad/tests/common/lightwalletd/send_transaction_test.rs

```diff
@@ -252,136 +245,36 @@ pub async fn run() -> Result<()> {
-    let partial_sync_height =
-        load_tip_height_from_state_directory(network, zebrad_state_path.as_ref()).await?;
-
-    tracing::info!(
-        ?partial_sync_height,
-        partial_sync_path = ?zebrad_state_path,
-        "performing full sync...",
-    );
-
-    let full_sync_path =
-        copy_state_and_perform_full_sync(network, zebrad_state_path.as_ref()).await?;
-
-    tracing::info!(?full_sync_path, "loading transactions...");
-
-    let transactions =
-        load_transactions_from_block_after(partial_sync_height, network, full_sync_path.as_ref())
-            .await?;
-
-    Ok(transactions)
-}
-
-/// Loads transactions from a block that's after the specified `height`.
+/// Loads transactions from 1-3 block(s) after the chain tip of the cached state.
 ///
-/// Starts at the block after the block at the specified `height`, and stops when it finds a block
-/// from where it can load at least one non-coinbase transaction.
+/// Returns a list of valid transactions that are not in any of the blocks finalized to disk
+/// in the original `zebrad_state_path`.
 ///
-/// # Panics
+/// ## Panics
 ///
-/// If the specified `zebrad_state_path` contains a chain state that's not synchronized to a tip that's
-/// after `height`.
+/// If the provided `test_type` doesn't need an rpc server and cached state
 #[tracing::instrument]
-async fn load_transactions_from_block_after(
-    height: block::Height,
+async fn load_transactions_from_future_blocks(
     network: Network,
-    zebrad_state_path: &Path,
+    test_type: TestType,
+    test_name: &str,
 ) -> Result<Vec<Arc<Transaction>>> {
-    let (_read_write_state_service, mut state, latest_chain_tip, _chain_tip_change) =
-        start_state_service_with_cache_dir(network, zebrad_state_path).await?;
-
-    let tip_height = latest_chain_tip
-        .best_tip_height()
-        .ok_or_else(|| eyre!("State directory doesn't have a chain tip block"))?;
-
-    assert!(
-        tip_height > height,
-        "Chain not synchronized to a block after the specified height",
-    );
-
-    let mut target_height = height.0;
-    let mut transactions = Vec::new();
-
-    while transactions.len() < max_sent_transactions() {
-        let new_transactions =
-            load_transactions_from_block(block::Height(target_height), &mut state).await?;
-
-        if let Some(mut new_transactions) = new_transactions {
-            new_transactions.retain(|transaction| !transaction.is_coinbase());
-            transactions.append(&mut new_transactions);
-        } else {
-            tracing::info!(
-                "Reached the end of the finalized chain\n\
-                 collected {} transactions from {} blocks before {target_height:?}",
-                transactions.len(),
-                target_height - height.0 - 1,
-            );
-            break;
-        }
-
-        target_height += 1;
-    }
+    let transactions = get_future_blocks(network, test_type, test_name, 3)
```

Comment:
Is 3 blocks always going to be enough to get 10 non-coinbase transactions?
Sometimes if the network is quiet, there will be a few blocks in a row without any non-coinbase transactions.

I'd suggest something like 20 or 50 blocks.

Nitpick: Can you make the number of blocks into a constant at the top of the module?

Reply:
Moved to a constant and updated to 50 blocks in [`d72ea40`](https://github.com/ZcashFoundation/zebra/pull/5589/commits/d72ea40650530d1dfa5e3bdd73d9d146eb28fba8)